core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2062,8 +2062,10 @@ private[spark] object Utils extends Logging {
     try {
       val properties = new Properties()
       properties.load(inReader)
-      properties.stringPropertyNames().asScala.map(
-        k => (k, properties.getProperty(k).trim)).toMap
+      properties.stringPropertyNames().asScala
+        .map(k => (k, properties.getProperty(k)))
```

Comment:
Isn't this going to break existing apps dependent on trimmed values?

Reply:
@gerashegalov would you please elaborate the use case here? I saw that you're treating `\n` as a property value, what is the specific usage scenario here?

Reply:
@HyukjinKwon your concern is valid although it has a simple solution of fixing up the file to your liking. Moreover the user editing a file for --properties-file probably legitimately expects the format prescribed by JDK https://docs.oracle.com/javase/6/docs/api/java/util/Properties.html#load(java.io.Reader)

@jerryshao the use case described in the JIRA is that our customers sometimes have some unusual line delimiters that we need to pass to Hadoop's `TextInputFormat`. In this case it's actually a conventional unix line separator and suppose the customer really insists that only `'\n'` should be the line separator and not the default set `["\n", "\r\n", "\r"]` . We had no issues with configuring it via Spark until we switched from `--conf` to the `--properties-file` to reduce the command line length.

Reply:
We could deprecate trimming, and have some other config to disable it but I think because it seems like a real bug to have disparity with --conf maybe we can just fix and document it?

Reply:
The changes here will break the current assumptions. Some editors will leave the trailing WS without removing it, but in fact user doesn't need that trailing WS, the changes here will break the assumptions, user have to check and remove all the trailing WS to avoid unexpected things.

AFAIK in Hive usually it uses ASCII or others to specify the separator, not "\n" or "\r\n", which will be removed or converted during the parse (which is quite brittle). So this is more like things you could fix in your side, not necessary in Spark side.

Reply:
>trim removes leading spaces as well that are totally legit.

It is hard to say which solution is legit, the way you proposed may be valid in your case, but it will be unexpected in other user's case. I'm not talking about legit or not, what I'm trying to say is that your proposal will break the convention, that's what I concerned about.

By ASCII I'm you can pass in ASCII number, and translate to actual char in the code, that will mitigate the problem here.

Reply:
I have also mentioned that we can make this a conditional logic: https://github.com/apache/spark/pull/22213#discussion_r212696259

Reply:
> By ASCII I mean you can pass in ASCII number, and translate to actual char in the code, that will mitigate the problem here.

I think I'll just keep passing the delimiter via `--conf` to Hadoop and everything else in a single properties to avoid dealing with manual conversion of ints to char.